multi-channel-chat/nats-channel/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use chatlog::*;
+use outbound::*;
+use serde::{Deserialize, Serialize};
+use wasmbus_rpc::actor::prelude::*;
+use wasmcloud_interface_messaging::*;
+use wasmcloud_interface_numbergen::*;
+
+#[allow(dead_code)]
+mod chatlog;
+
+#[allow(dead_code)]
+mod outbound;
+
+const CHANNEL_NAME: &str = "nats";
+const API_ACTOR: &str = "mcchat/api";
+const RTCHAT_TOPIC: &str = "rtchat.messages";
+
+#[derive(Debug, Default, Actor, HealthResponder)]
+#[services(Actor, MessageSubscriber, Outbound)]
+struct NatsChannelActor {}
+
+#[async_trait]
+impl Outbound for NatsChannelActor {
+    async fn publish_message(&self, ctx: &Context, arg: &OutboundMessage) -> RpcResult<bool> {
+        let pubber = MessagingSender::new();
+        let res = pubber
+            .publish(
+                ctx,
+                &PubMessage {
+                    subject: RTCHAT_TOPIC.to_string(),
+                    reply_to: None,
+                    body: serde_json::to_vec(arg).unwrap(),
+                },
+            )
+            .await;
+        Ok(res.is_ok())
+    }
+}
+
+/// Implementation of Chat Log actor trait methods
+#[async_trait]
+impl MessageSubscriber for NatsChannelActor {
+    async fn handle_message(&self, ctx: &Context, msg: &SubMessage) -> RpcResult<()> {
+        let reply_topic = msg
```

Comment:
This `reply_topic` is only ever borrowed, but this whole statement allocates twice in the case where it is `None`. Plus it would be hard to follow for someone new to rust. Consider replacing with

```rust
let reply_topic = msg.reply_to.as_deref().unwrap_or("dedletter")
```

Then you don't even need to borrow below as the type will be `&str`